librad/src/git/trailer.rs

```diff
@@ -54,6 +61,18 @@ impl<'a> Trailer<'a> {
     }
 }
 
+fn token_parser(s: &str) -> IResult<&str, &str> {
+    take_while(|c: char| c.is_alphanumeric() || c == '-')(s)
+}
+
+fn separator_parser(s: &str) -> IResult<&str, char> {
+    delimited(space0, nom::character::complete::char(':'), space0)(s)
```

Comment:
A slight nit: Parsing spaces here could be considered _too much_ work for a combinator. A separator parser would expect the separator and nothing else.
The same for `value_parser`, where it should be `zero_or_more(any_char)`. 

Then all the white space is dealt with a level above for a single trailer.

Reply:
You mean that a parser should be more granular, while here we are actually composing three parsers in one, right? Just to make sure I am getting your point straight :)

Reply:
Ya exactly. The `separator_parser` should only know how to parse the separator and not whitespace.

The thing that cares about white space is the trailer :)

Reply:
Sure thing :) First I want to figure the harder bits and then tweak all those nits :ok_hand:

Reply:
Sounds good :100:

Reply:
So back to this one, I honestly don't see the benefit of breaking this further now. The notion of separator _in this context_ is one of a given set of characters being present amongst possible one or more spaces.  Moving the part of the space higher up just increases complexity and detail where it is least needed (in this case) IMHO.

Reply:
Sure, ya that makes sense :+1: